Source/DafnyLanguageServer.Test/Unit/ParserExceptionTest.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Boogie;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Unit {
+
+  [TestClass]
+  public class ParserExceptionTest {
+    private static readonly string TestFilePath = "parserException.dfy";
+    private const string LanguageId = "dafny";
+    private const int MaxTestExecutionTimeMs = 10000;
+    private DafnyLangParser parser;
+    private LastDebugLogger lastDebugLogger;
+
+    [TestInitialize]
+    public void SetUp() {
+      lastDebugLogger = new LastDebugLogger();
+      parser = DafnyLangParser.Create(lastDebugLogger);
+    }
+
+    [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+    public void DocumentWithParserExceptionDisplaysIt() {
+      var source = "function t() { / }";
+      var documentItem = CreateTestDocument(source, TestFilePath);
+      var errorReporter = new ParserExceptionSimulatingErrorReporter();
+      parser.Parse(documentItem, errorReporter, new CancellationToken());
```

Comment:
You can use `default` in place of `new CancellationToken()` since it's a `struct` type.